src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesImplementation.cs

```diff
@@ -69,19 +80,16 @@ protected sealed override void Dispose(bool disposing)
             PaddingMode paddingMode,
             out int bytesWritten)
         {
-            ILiteSymmetricCipher cipher = CreateLiteCipher(
+            return OneShotTransformation(
+                ref _encryptEcbLiteHash,
                 CipherMode.ECB,
-                Key,
                 iv: default,
-                blockSize: BlockSize / BitsPerByte,
-                paddingSize: BlockSize / BitsPerByte,
-                0, /*feedback size */
-                encrypting: true);
-
-            using (cipher)
-            {
-                return UniversalCryptoOneShot.OneShotEncrypt(cipher, paddingMode, plaintext, destination, out bytesWritten);
-            }
+                encrypting: true,
+                paddingMode,
+                plaintext,
+                destination,
+                UniversalCryptoOneShot.OneShotEncrypt,
+                out bytesWritten);
         }
```

Comment:
This pattern repeats over, and over, and over.  (12 times, if my math's right)  Is there any way we can reduce the copy-pasta?  (The best I've come up with is to make a class `CachedKeyState` and defer these methods into _cachedKey.TryEncryptEcbCore (et al)... but that doesn't feel obviously better)

Reply:
It's a method call but they all pass in different parameters between encrypting, mode, etc. The introduction of `OneShotTransformation` _was_ my attempt at reducing duplication.

On top of this relying heavily on partials to avoid every allocation as possible, I am inclined to say "I think this looks fine". I also don't know that we want to burn an allocation on an abstraction for otherwise readable code.